plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -264,8 +265,28 @@ func getHTTPRoundTripper(c *config.Configuration, logger *zap.Logger) (rt http.R
 		TLSHandshakeTimeout: 10 * time.Second,
 		TLSClientConfig:     ctlsConfig,
 	}
+
+	token := ""
+	if c.TokenFilePath != "" {
+		tokenFromFile, err := loadToken(c.TokenFilePath)
+		if err != nil {
+			return nil, err
+		}
+		token = tokenFromFile
+	}
 	return bearertoken.RoundTripper{
 		Transport:       httpTransport,
 		OverrideFromCtx: true,
+		StaticToken:     token,
 	}, nil
 }
+
+func loadToken(path string) (string, error) {
+	b, err := os.ReadFile(filepath.Clean(path))
+	if err != nil {
+		// wrapping the error message
+		err = fmt.Errorf("failed to get token from file: %w", err)
+		return "", err
```

Comment:
unnecessary verbosity
```suggestion
		return "", fmt.Errorf("failed to get token from file: %w", err)
```